### PR TITLE
arch: x86_64: enable HTT flag

### PR DESCRIPTION
**Pull request by thomasbarrett**
When the HTT flag CPUID.1.EDX[HTT] is 0, it indicates that there is only a single logical processor in the package. When HTT is 1, it indicates that CPUID.1.EBX[23:16] contains the number of logical processors in the package.

When this information is not included in CPUID leaf 0x1, some cpu topology enumeration software such as hwloc are known to crash.

Note: 
I have experienced crashes on the version of lstopo available on Ubuntu 22.04 on some (but interestingly not all) machines I've tested on. Newer version of lstopo [work around this bug](https://github.com/open-mpi/hwloc/commit/33b555b5a1c8339daeb4215bad57430b57a6b33f), but its still good to be correct for maximum compatability.

TIL that the HTT flag has [absolutely nothing](https://www.os2museum.com/wp/htt-means-hyper-threading-right/) to do with hyper-threading 😆.

### CODE REVIEW

**Review by rbradford — APPROVED** [review]

File: arch/src/x86_64/mod.rs [hunk 1]
```diff
@@ -1277,6 +1277,14 @@ fn update_cpuid_topology(
     let core_width = (8 - (cores_per_die - 1).leading_zeros()) + thread_width;
     let die_width = (8 - (dies_per_package - 1).leading_zeros()) + core_width;
 
+    let mut cpu_ebx = CpuidPatch::get_cpuid_reg(cpuid, 0x1, None, CpuidReg::EBX).unwrap_or(0);
+    cpu_ebx |= u32::from(dies_per_package * cores_per_die * threads_per_core) << 16;
```
Comment (likebreath) [inline arch/src/x86_64/mod.rs:1281]: Should this be a "u8" if you are only overwriting `EBX[23:16]`?

Comment (rbradford) [reply arch/src/x86_64/mod.rs:1281]: I think it needs to be u32 since you can only bitwise with identical types in Rust and `cpu_ebx` is u32.

Comment (likebreath) [reply arch/src/x86_64/mod.rs:1281]: I was worried about overwriting more than 8 bits. When I looked further, I found the overwriting is not an issue as an overflow would happen first from `dies_per_package * cores_per_die * threads_per_core`. And this is not the only place where we will overflow from such multiplication. That's a separate issue to fix.

Comment (rbradford) [reply arch/src/x86_64/mod.rs:1281]: I now see what you mean, indeed it should be masked like this:

```cpu_ebx |= u32::from((dies_per_package * cores_per_die * threads_per_core)  & 0xff) << 16```

Comment (thomasbarrett) [reply arch/src/x86_64/mod.rs:1281]: Ack. Updating.

**Review by likebreath — APPROVED** [review]

**Pull request merged**